target_cratedb/connector.py

```diff
@@ -36,6 +39,34 @@ def create_engine(self) -> sa.Engine:
         polyfill_refresh_after_dml_engine(engine)
         return engine
 
+    @contextmanager
+    def _connect(self) -> t.Iterator[sa.engine.Connection]:
+        """
+        Connect to the database.
+
+        Note: Needs to be overwritten to perform a CrateDB version check.
+        """
+        engine = self._engine
+        with engine.connect().execution_options() as conn:
+            self._check_cratedb_620(conn)
+            yield conn
+        engine.dispose()
+
+    def _check_cratedb_620(self, conn: sa.Connection):
+        """
+        Fail if the CrateDB version is lower than 6.2.
+
+        -- https://github.com/crate/crate/issues/15161
+        """
+        with conn.begin():
+            version_raw = conn.exec_driver_sql("SELECT version()").scalar_one()
+        matches = re.match(r"^CrateDB (\d+\.\d+\.\d+).*", version_raw)
```

Comment:
idea: you could perhaps avoid regex/parsing by querying directly `select version['number'] from sys.nodes` that returns version